### PR TITLE
Update Add command to check for existing Ic Number

### PR DESCRIPTION
**Pull request by AaronJT1**
# Summary of change
Fixed bugs for DeleteCommand and Edit command and update add command to check for existing ic number
---
# Key changes

## `DeleteCommand.java` and `EditCommand.java`
1. create getCurrentPatientList method in ModelManager.java to get list of current patients
instead of using the getFilteredPatientList

## Update Add Command
1. Created isPatientWithIcNumberPresent method which checks if there is an existing patient with that ic number
2. Updated the AddCommandTest to test this method

### CODE REVIEW

File: src/test/java/seedu/address/logic/commands/AddCommandTest.java [hunk 1]
```diff
@@ -55,6 +59,17 @@ public void execute_duplicatePatient_throwsCommandException() {
         assertThrows(CommandException.class, AddCommand.MESSAGE_DUPLICATE_PATIENT, () -> addCommand.execute(modelStub));
     }
 
+    @Test
```
Comment (longnguyentan) [inline src/test/java/seedu/address/logic/commands/AddCommandTest.java:62]: Good efforts in testing :)

**Review by longnguyentan — APPROVED** [review]

File: src/main/java/seedu/address/logic/commands/AddCommand.java [hunk 2]
```diff
@@ -56,7 +61,9 @@ public CommandResult execute(Model model) throws CommandException {
         if (model.hasPatient(toAdd)) {
             throw new CommandException(MESSAGE_DUPLICATE_PATIENT);
         }
-
+        if (model.isPatientWithIcNumberPresent(toAdd.getIcNumber())) {
+            throw new CommandException(MESSAGE_PATIENT_WITH_IC_NUMBER_ALREADY_EXIST);
+        }
```
Comment (thienmy0) [inline src/main/java/seedu/address/logic/commands/AddCommand.java:66]: I like how you check if the Patient is already in the System!

File: src/main/java/seedu/address/model/Model.java [hunk 3]
```diff
@@ -26,14 +26,21 @@ public interface Model {
...
+     * Returns the patient with the IcNumber from patientList
+     */
     Patient getPatient(IcNumber icNumber, List<Patient> patientList);
-    /*public Patient getPatient(IcNumber icNumber, List<Patient> patientList){
-        for (int i = 0; i < patientList.size(); i++) {
-            if(patientList.get(i).getIcNumber().equals(icNumber))
-                return patientList.get(i);
-        }
-        return null;
-    }*/
```
Comment (thienmy0) [inline src/main/java/seedu/address/model/Model.java:36]: Nice to see you removing unused code!

File: src/main/java/seedu/address/model/ModelManager.java [hunk 4]
```diff
@@ -15,6 +15,8 @@
 import seedu.address.model.patient.IcNumber;
 import seedu.address.model.patient.Patient;
 
+import javax.swing.*;
```
Comment (thienmy0) [inline src/main/java/seedu/address/model/ModelManager.java:18]: Should this import statement be a * ?

**Review by thienmy0 — APPROVED** [review]
Looks good to me, good job!

**Pull request merged**